Creational/TelecommunicationFactory/ImsCall.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * File ImsCall.php
+ * @package design-patterns
+ * @author pn
+ * @version 1.0
+ */
+namespace DesignPatterns\Creational\TelecommunicationFactory;
+
+/**
+ * Class ImsCall
+ *
+ * @author Phat Ngyen <phatnt2721989@gmail.com>
+ * @package DesignPatterns\Creational\TelecommunicationFactory
+ */
+class ImsCall extends Call
+{
+  const IMS_COST = 4;
+
+  /**
+   * @return string
+   */
+  public function handleCall()
+  {
+    return str_split($this->getCall(), 1);
+  }
+
+  /**
+   * @param $time
+   * @return int
+   */
+  public function chargeCall(int $time)
+  {
+    return $time * IMS_COST;
```

Comment:
Should create an abstract method named `getCost` on `Call` class, and move the chargeCall to the parent class. That will reduce the work that the sub classes need to implement